### DISCUSSION TITLE
How can I reset a radio button when using Formik?

### DISCUSSION THREAD

**Discussion by MikeWillis** [chakra-ui/chakra-ui#6532] [Q&A]
I have a couple of required `RadioGroup` sets in my form. Everything is working fine as far as filling them out, hitting submit, sending the data etc. But after submit I call Formik's `actions.resetForm()`, which clears all fields, but leaves the checked radio inputs checked.

Now the fields will be visually checked, but in reality their state is reset to the default, which is empty. However, the user can still hit submit and validation will not be triggered.

The only thing I've come up with so far is to re-render the form entirely. It's ugly though and seems hacky. Has anyone tackled this before and come up with better solution?

**Comment by alsomariya** [comment]
Hi, were you able to solve this? Facing same issue.

**Reply by MikeWillis** [reply]
unfortunately no, although I haven't touched that project too much since I first asked this question, so maybe there are better solutions now.. The solution I came up with works, I just don't like it much..

I basically have something like this:

    let fields = [
        {
            name: 'whatever',
            label: 'Whatever',
            required: true,
            type: "radio",
            options: [
                { text: "Option A" },
                { text: "Option B" },
                { text: "Option C" },
            ]
        },
        // more fields here
    ];

    const MyComponent = props => {
        const [st_fields,sst_fields] = useState(fields);
        
        // just a function that renders my fields, nothing special
        let renderFields = ()=>{
                return st_fields.map(field=>{});
        };

        let handleSubmit = async (values,actions) => {
            // do stuff to submit the form with axios or fetch or whatever

            // if all is good above, I reset the form like so:
            actions.resetForm();
            actions.setSubmitting(false);

            // now blank out the fields, wait a tiny bit of time, and reset the fields
            // this will trigger a re-render, and the radio buttons should be fully reset
            sst_fields([]);
            setTimeout(()=>{
                sst_fields(fields);
            }, 1);
            window.parent.scrollTo(0,0);
        }; // handleSubmit
        
        return (
            <Box>
                <Formik
                    initialValues={initialValues}
                    onSubmit={handleSubmit}
                >
                    {props => {
                        return (
                            <Form>
                                { renderFields(st_fields) }
                                <Center>
                                    <Button
                                        isLoading={props.isSubmitting}
                                        type='submit'
                                    >
                                        Send
                                    </Button>
                                </Center>
                            </Form>
                        );
                    }}
                </Formik>
            </Box>
        );
    }; // MyComponent

    export default MyComponent;

**Comment by sumeet030990** [comment]
facing same issue
